diff -urNad libwebp-orig/makefile webp/libwebp/makefile
--- libwebp-orig/makefile	2010-09-24 07:31:28.000000000 +0300
+++ libwebp/makefile	2010-10-17 11:18:12.000000000 +0300
@@ -78,7 +78,7 @@
 
 ######################################################################
 
-all:	$(SRC:%.c=%)
+all: libwebp.so
 
 
 ######################################################################
@@ -86,10 +86,19 @@
 webpconv: libwebp.o webpconv.o $(LEPTLIB)
 	$(CC) -o webpconv webpconv.o $(ALL_LIBS) libwebp.o $(VPX_LIB)
 
-libwebp.o: webpimg.c
-	$(CC) -c $(CPPFLAGS) -o libwebp.o webpimg.c
+webpimg.o: webpimg.c
+	$(CC) $(CFLAGS) -o $@ -c $<
+
+libwebp.so: webpimg.o
+	$(CC) -shared -o $@ $< $(VPX_LIB) -Wl,-soname -Wl,$@
+
+install:
+	mkdir -p $(DESTDIR)$(prefix)/include
+	install -m 644 webpimg.h $(DESTDIR)$(prefix)/include/webpimg.h
+	mkdir -p $(DESTDIR)$(prefix)/lib
+	install -m 755 libwebp.so $(DESTDIR)$(prefix)/lib/libwebp.so
 
 clean:
-	rm webpconv *.o
+	rm -r webpconv *.o *.so
 
 ######################################################################
